doc/source/schedule-backup.rst

```diff
@@ -0,0 +1,51 @@
+.. _schedule: 
+
+Schedule backups
+==============================
+
+In |PBM| version 1.4.1 and earlier, we recommend using ``crond`` or similar services to schedule backup snapshots.
+
+.. important::
+
+   Before configuring ``crond``, make sure that you have :ref:`installed <install>` and :ref:`configured <initial-setup>` |PBM| to make backups in your database. Start a backup manually to verify this: :command:`pbm backup`. 
+
+The recommended approach is to create a :file:`crontab` file in the :dir:`/ect/cron.d` directory and specify the command in it. This simplifies server administration especially if multiple users have access to it.
+
+As an example, let's configure to run backup snapshots on 23:30 every Sunday. 
+
+The steps are the following:
+
+1. Create a ``crontab`` file. Let's name it ``pbm-backup``. 
+2. Specify the command in the file:
+   
+   .. code-block:: guess
+   
+      30 23 * * sun . /etc/default/pbm-agent; /usr/bin/pbm backup 
+
+ where the :file:`/etc/default/pbm-agent` is the environment variable file path for Debian and Ubuntu and the dot ``.`` equals to bash builtin ``source``.
```

Comment:
```suggestion
 Please note the dot ``.`` before /etc/default/pbm-agent - it sources (includes) the file for the rest of the shell commands being run. This allows you to put the "export PBM_MONGODB_URI=mongodb://username:password@host1,.../" statement in a file and have that variable in the ``pbm`` program's environment rather than supplied as a command line argument, which is insecure. 
 
`/etc/default/pbm-agent` is the default location used for the environment variable for ``pbm-agent`` services. This is where the PBM_MONGODB_URI variable is usually set. You can use another path if you like to, however.
```

Reply:
IMHO, using /etc/default/pbm-agent (or /etc/sysconfig/pbm-agent file for RPM based systems) isn't correct due to several reasons:
1) since it's default location for the env variable for pbm-agent service, it shouldn't contain 'export' command, the proper syntax is VARIABLE=VALUE
2) without export command env variable won't be available for pbm process during cron command execution

I think there are several ways to do this:
1) if it's really required to use default env file, cron command should be like this
30 23 * * sun root . /etc/default/pbm-agent; PBM_MONGODB_URI=$PBM_MONGODB_URI /usr/bin/pbm backup

BTW, if to create cron task under /etc/cron.d is necessary to set user (used to execute command) and in this case root is required since /etc/default/pbm-agent (or /etc/sysconfig/pbm-agent) is owned by root

2) create a separate env file (let's say /etc/default/pbm-cron) with syntax like
export PBM_MONGODB_URI="....."
then cron command should look like
30 23 * * sun any_user_with_access_to_env_file . /etc/default/pbm-cron; /usr/bin/pbm backup

Reply:
@akira-kurogane Which variant should we describe?

Reply:
Hi @sandraromanchenko 

You're right - the /etc/default/pbm-agent file, consumed by systemd in it's service-launching procedure, is the "VAR=value" syntax. This clashes with the "export VAR=value" format it would have to be as a shell include file, for the crond command line. I was lax in that I didn't install pbm-agent properly and went from memory as to what was in the file. My apologies that I've sent both of you on this wild goose chase. Also I was wrong in thinking that su'ing as root was avoidable if using the /etc/default/pbm-agent file.

I like the (1) method shown above:
`30 23 * * sun root . /etc/default/pbm-agent; PBM_MONGODB_URI=$PBM_MONGODB_URI /usr/bin/pbm backup`

 which brings the "VAR=value" /etc/default/pbm-agent file's value into the environment of the pbm CLI process being launched. Without causing it to be in the command line arguments that other users can see if they run 'ps -ef' etc, which is the original security problem we're trying to overcome.

But there will be users who don't have the /etc/default/pbm-agent file on the server they run this pbm CLI crond command from and/or who don't want to run as root, and they should use (2). So we'll have to document both methods I think.